contracts/colonyNetwork/ColonyNetworkDeployer.sol

```diff
@@ -126,11 +127,69 @@ contract ColonyNetworkDeployer is ColonyNetworkStorage {
     return (address(token), colonyAddress);
   }
 
+  /**
+   * @dev Generates pseudo-randomly a salt value using a diverse selection of block and
+   * transaction properties.
+   * @return salt The 32-byte pseudo-random salt value.
+   */
+  function getColonyCreationSalt() public view returns (bytes32 salt) {
+    // We only want Colony Networks to be able to deploy to the same address,
+    // so we use the permissioned deploy protection feature of CreateX, and set
+    // the first 160 bits of the salt to the address of this contract.
+
+    salt = bytes32(uint256(uint160(address(this)))) << 96;
+    bytes32 additionalSalt;
+    // This portion of the function has been adapted from the CreateX library licensed under AGPL-3.0-only
+    // NB it is only pseudo-random but that is sufficient for our purposes
+    // The fact we are doing this means that a caller cannot make two colonies in the same transaction
+    unchecked {
+      additionalSalt = keccak256(
+        abi.encode(
+          // We don't use `block.number - 256` (the maximum value on the EVM) to accommodate
```

Comment:
I'd be comfortable linking to the comment in the CreateX source code: https://github.com/pcaversaccio/createx/blob/main/src/CreateX.sol#L965

Reply:
I'm wary of doing so in case the repository disappears or its structure changes.